spec/controllers/passengers_controller_spec.rb

```diff
@@ -9,91 +9,84 @@
     session[:user_id] = admin.id
   end
 
+  let!(:passenger) { create :passenger, :permanent }
+
   describe 'POST set_status' do
     context 'with broken mailer' do
+      before :each do
+        allow(PassengerMailer).to receive(:notify_archived).and_raise Net::SMTPFatalError
+        allow(PassengerMailer).to receive(:notify_active).and_raise Net::SMTPFatalError
+      end
       context 'changing active to archived' do
         it 'saves the passenger and displays a warning message' do
-          allow(PassengerMailer).to receive(:notify_archived).and_raise Net::SMTPFatalError
-
-          @passenger = create :passenger, :permanent, active_status: 'active'
-          post :set_status, params: { id: @passenger.id, status: 'archived' }
-          @passenger.reload
+          passenger.update_columns active_status: 'active'
+          post :set_status, params: { id: passenger.id, status: 'archived' }
+          passenger.reload
 
-          expect(@passenger.active_status).to eq 'archived'
+          expect(passenger.active_status).to eq 'archived'
           expect(response).to redirect_to(passengers_path)
-          expect(flash[:warning]).to match(/the passenger was unable to be notified of their status change via email/)
+          expect(flash[:warning]).to match(Regexp.union(/could not notify/, /status change/))
```

Comment:
`Regexp.union` (more or less) creates an alternation. That is, this is the same as `/could not notify|status change/`, which will match if either phase is present in the string. I suspect you wanted _both_. If so, the easiest way is probably two expectations.